src/fswAlgorithms/attGuidance/flybyPoint/flybyPointAlgorithm.cpp

```diff
@@ -0,0 +1,219 @@
+/*
+ ISC License
+
+ Copyright (c) 2025, Laboratory for Atmospheric and Space Physics, University of Colorado at Boulder
+
+ Permission to use, copy, modify, and/or distribute this software for any
+ purpose with or without fee is hereby granted, provided that the above
+ copyright notice and this permission notice appear in all copies.
+
+ THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+ WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+ MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+ ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+ WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+ ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+ OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+
+ */
+
+#include "fswAlgorithms/attGuidance/flybyPoint/flybyPointAlgorithm.h"
+#include "architecture/utilities/avsEigenSupport.h"
+#include "architecture/utilities/macroDefinitions.h"
+#include "architecture/utilities/rigidBodyKinematics.hpp"
+
+/*! This method is used to reset the module.
+ @return void
+ */
+void FlybyPointAlgorithm::reset(uint64_t currentSimNanos) {
+    this->lastFilterReadTime = 0;
+    this->firstRead = true;
+}
+
+/*! This method is the main carrier for the boresight calculation routine.  If it detects
+ that it needs to re-init (direction change maybe) it will re-init itself.
+ Then it will compute the angles away that the boresight is from the celestial target.
+ @return void
+ @param currentSimNanos The current simulation time for system
+ */
+AttRefMsgPayload FlybyPointAlgorithm::updateState(uint64_t currentSimNanos,
```

Comment:
:small_red_triangle_down: There are a lot of places where we could add `const` throughout this algorithm. I'm happy to help you set up your IDE such that it shows you where.